// react-mazer-ui/dist/Alert.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.AlertDismissable = exports.AlertHeading = exports.AlertLightInfo = exports.AlertLightLight = exports.AlertLightSecondary = exports.AlertLightDark = exports.AlertLightDanger = exports.AlertLightWarning = exports.AlertLightSuccess = exports.AlertLightPrimary = exports.AlertInfo = exports.AlertLight = exports.AlertSecondary = exports.AlertDark = exports.AlertDanger = exports.AlertWarning = exports.AlertSuccess = exports.AlertPrimary = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _Alert = _interopRequireDefault(require("./Alert"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'Example/Alert',
  component: _Alert.default,
  argTypes: {}
};
exports.default = _default;

var Template = function Template(args) {
  return /*#__PURE__*/_react.default.createElement(_Alert.default, args);
};

var AlertPrimary = Template.bind({});
exports.AlertPrimary = AlertPrimary;
AlertPrimary.args = {
  content: "This is primary alert.",
  status: "primary"
};
var AlertSuccess = Template.bind({});
exports.AlertSuccess = AlertSuccess;
AlertSuccess.args = {
  content: "This is success alert.",
  status: "success"
};
var AlertWarning = Template.bind({});
exports.AlertWarning = AlertWarning;
AlertWarning.args = {
  content: "This is warning alert.",
  status: "warning"
};
var AlertDanger = Template.bind({});
exports.AlertDanger = AlertDanger;
AlertDanger.args = {
  content: "This is danger alert.",
  status: "danger"
};
var AlertDark = Template.bind({});
exports.AlertDark = AlertDark;
AlertDark.args = {
  content: "This is dark alert.",
  status: "dark"
};
var AlertSecondary = Template.bind({});
exports.AlertSecondary = AlertSecondary;
AlertSecondary.args = {
  content: "This is secondary alert.",
  status: "secondary"
};
var AlertLight = Template.bind({});
exports.AlertLight = AlertLight;
AlertLight.args = {
  content: "This is light alert.",
  status: "light"
};
var AlertInfo = Template.bind({});
exports.AlertInfo = AlertInfo;
AlertInfo.args = {
  content: "This is info alert.",
  status: "info"
}; // ---

var AlertLightPrimary = Template.bind({});
exports.AlertLightPrimary = AlertLightPrimary;
AlertLightPrimary.args = {
  content: "This is primary alert.",
  status: "primary",
  light: true
};
var AlertLightSuccess = Template.bind({});
exports.AlertLightSuccess = AlertLightSuccess;
AlertLightSuccess.args = {
  content: "This is success alert.",
  status: "success",
  light: true
};
var AlertLightWarning = Template.bind({});
exports.AlertLightWarning = AlertLightWarning;
AlertLightWarning.args = {
  content: "This is warning alert.",
  status: "warning",
  light: true
};
var AlertLightDanger = Template.bind({});
exports.AlertLightDanger = AlertLightDanger;
AlertLightDanger.args = {
  content: "This is danger alert.",
  status: "danger",
  light: true
};
var AlertLightDark = Template.bind({});
exports.AlertLightDark = AlertLightDark;
AlertLightDark.args = {
  content: "This is dark alert.",
  status: "dark",
  light: true
};
var AlertLightSecondary = Template.bind({});
exports.AlertLightSecondary = AlertLightSecondary;
AlertLightSecondary.args = {
  content: "This is secondary alert.",
  status: "secondary",
  light: true
};
var AlertLightLight = Template.bind({});
exports.AlertLightLight = AlertLightLight;
AlertLightLight.args = {
  content: "This is light alert.",
  status: "light",
  light: true
};
var AlertLightInfo = Template.bind({});
exports.AlertLightInfo = AlertLightInfo;
AlertLightInfo.args = {
  content: "This is info alert.",
  status: "info",
  light: true
};
var AlertHeading = Template.bind({});
exports.AlertHeading = AlertHeading;
AlertHeading.args = {
  heading: "Primary",
  content: "This is primary alert.",
  status: "primary"
};
var AlertDismissable = Template.bind({});
exports.AlertDismissable = AlertDismissable;
AlertDismissable.args = {
  content: "This is primary alert.",
  status: "primary",
  toggle: function toggle() {
    return alert("Close it");
  },
  isOpen: true
};
