// react-mazer-ui/dist/Alert.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = Alert;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

require("bootstrap/js/dist/alert");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function Alert(props) {
  var status = props.status,
      heading = props.heading,
      content = props.content,
      light = props.light,
      isOpen = props.isOpen,
      toggle = props.toggle;

  var getClassStatus = function getClassStatus(val) {
    return val ? "alert-light-".concat(status) : "alert-".concat(status);
  };

  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, typeof toggle == "function" ? /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, isOpen ? /*#__PURE__*/_react.default.createElement("div", {
    className: "alert ".concat(getClassStatus(light), " alert-dismissible fade show"),
    role: "alert"
  }, heading ? /*#__PURE__*/_react.default.createElement("h4", {
    className: "alert-heading"
  }, heading) : null, content ? /*#__PURE__*/_react.default.createElement("p", null, content) : null, /*#__PURE__*/_react.default.createElement("button", {
    type: "button",
    className: "btn-close",
    "aria-label": "Close",
    onClick: toggle
  })) : null) : /*#__PURE__*/_react.default.createElement("div", {
    className: "alert ".concat(getClassStatus(light)),
    role: "alert"
  }, heading ? /*#__PURE__*/_react.default.createElement("h4", {
    className: "alert-heading"
  }, heading) : null, content ? /*#__PURE__*/_react.default.createElement("p", null, content) : null));
}

Alert.propTypes = {
  status: _propTypes.default.string,
  heading: _propTypes.default.string,
  content: _propTypes.default.string,
  light: _propTypes.default.bool,
  isOpen: _propTypes.default.bool,
  toggle: _propTypes.default.func
};
Alert.defaultProps = {
  status: "primary",
  heading: "",
  content: "",
  light: false,
  isOpen: true,
  toggle: false
};
