// react-mazer-ui/dist/Badge.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.BadgeInfoLight = exports.BadgeWarningLight = exports.BadgeDangerLight = exports.BadgeSecondaryLight = exports.BadgePrimaryLight = exports.BadgeDark = exports.BadgeInfo = exports.BadgeWarning = exports.BadgeDanger = exports.BadgeSecondary = exports.BadgePrimary = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _Badge = _interopRequireDefault(require("./Badge"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'Example/Badge',
  component: _Badge.default,
  argTypes: {}
};
exports.default = _default;

var Template = function Template(args) {
  return /*#__PURE__*/_react.default.createElement(_Badge.default, args);
};

var BadgePrimary = Template.bind({});
exports.BadgePrimary = BadgePrimary;
BadgePrimary.args = {
  label: "Primary",
  status: "primary"
};
var BadgeSecondary = Template.bind({});
exports.BadgeSecondary = BadgeSecondary;
BadgeSecondary.args = {
  label: "Secondary",
  status: "secondary"
};
var BadgeDanger = Template.bind({});
exports.BadgeDanger = BadgeDanger;
BadgeDanger.args = {
  label: "Danger",
  status: "danger"
};
var BadgeWarning = Template.bind({});
exports.BadgeWarning = BadgeWarning;
BadgeWarning.args = {
  label: "Warning",
  status: "warning"
};
var BadgeInfo = Template.bind({});
exports.BadgeInfo = BadgeInfo;
BadgeInfo.args = {
  label: "Info",
  status: "info"
};
var BadgeDark = Template.bind({});
exports.BadgeDark = BadgeDark;
BadgeDark.args = {
  label: "Dark",
  status: "dark"
}; // light

var BadgePrimaryLight = Template.bind({});
exports.BadgePrimaryLight = BadgePrimaryLight;
BadgePrimaryLight.args = {
  label: "Primary",
  status: "primary",
  light: true
};
var BadgeSecondaryLight = Template.bind({});
exports.BadgeSecondaryLight = BadgeSecondaryLight;
BadgeSecondaryLight.args = {
  label: "Secondary",
  status: "secondary",
  light: true
};
var BadgeDangerLight = Template.bind({});
exports.BadgeDangerLight = BadgeDangerLight;
BadgeDangerLight.args = {
  label: "Danger",
  status: "danger",
  light: true
};
var BadgeWarningLight = Template.bind({});
exports.BadgeWarningLight = BadgeWarningLight;
BadgeWarningLight.args = {
  label: "Warning",
  status: "warning",
  light: true
};
var BadgeInfoLight = Template.bind({});
exports.BadgeInfoLight = BadgeInfoLight;
BadgeInfoLight.args = {
  label: "Info",
  status: "info",
  light: true
};
