// react-mazer-ui/dist/Button.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ButtonGroup = exports.ButtonIcon = exports.ButtonDisabled = exports.ButtonLarge = exports.ButtonNormal = exports.ButtonSmall = exports.ButtonRounded = exports.DarkOutline = exports.LightOutline = exports.SuccessOutline = exports.DangerOutline = exports.WarningOutline = exports.InfoOutline = exports.SecondaryOutline = exports.PrimaryOutline = exports.Dark = exports.Light = exports.Success = exports.Danger = exports.Warning = exports.Info = exports.Secondary = exports.Primary = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _Button = _interopRequireDefault(require("./Button"));

var _ButtonGroup = _interopRequireDefault(require("./ButtonGroup"));

var _Icon = _interopRequireDefault(require("./Icon"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'Example/Button',
  component: _Button.default,
  argTypes: {}
};
exports.default = _default;

var Template = function Template(args) {
  return /*#__PURE__*/_react.default.createElement(_Button.default, args);
};

var ButtonGroupWrapper = function ButtonGroupWrapper(args) {
  return /*#__PURE__*/_react.default.createElement(_ButtonGroup.default, args);
};

var Primary = Template.bind({});
exports.Primary = Primary;
Primary.args = {
  label: "Primary"
};
var Secondary = Template.bind({});
exports.Secondary = Secondary;
Secondary.args = {
  label: "Secondary",
  status: "secondary"
};
var Info = Template.bind({});
exports.Info = Info;
Info.args = {
  label: "Info",
  status: "info"
};
var Warning = Template.bind({});
exports.Warning = Warning;
Warning.args = {
  label: "Warning",
  status: "warning"
};
var Danger = Template.bind({});
exports.Danger = Danger;
Danger.args = {
  label: "Danger",
  status: "danger"
};
var Success = Template.bind({});
exports.Success = Success;
Success.args = {
  label: "Success",
  status: "success"
};
var Light = Template.bind({});
exports.Light = Light;
Light.args = {
  label: "Light",
  status: "light"
};
var Dark = Template.bind({});
exports.Dark = Dark;
Dark.args = {
  label: "Dark",
  status: "dark"
};
var PrimaryOutline = Template.bind({});
exports.PrimaryOutline = PrimaryOutline;
PrimaryOutline.args = {
  label: "Primary Outline",
  status: "primary-outline"
};
var SecondaryOutline = Template.bind({});
exports.SecondaryOutline = SecondaryOutline;
SecondaryOutline.args = {
  label: "Secondary Outline",
  status: "secondary-outline"
};
var InfoOutline = Template.bind({});
exports.InfoOutline = InfoOutline;
InfoOutline.args = {
  label: "Info Outline",
  status: "info-outline"
};
var WarningOutline = Template.bind({});
exports.WarningOutline = WarningOutline;
WarningOutline.args = {
  label: "Warning Outline",
  status: "warning-outline"
};
var DangerOutline = Template.bind({});
exports.DangerOutline = DangerOutline;
DangerOutline.args = {
  label: "Danger Outline",
  status: "danger-outline"
};
var SuccessOutline = Template.bind({});
exports.SuccessOutline = SuccessOutline;
SuccessOutline.args = {
  label: "Success Outline",
  status: "success-outline"
};
var LightOutline = Template.bind({});
exports.LightOutline = LightOutline;
LightOutline.args = {
  label: "Light Outline",
  status: "light-outline"
};
var DarkOutline = Template.bind({});
exports.DarkOutline = DarkOutline;
DarkOutline.args = {
  label: "Dark Outline",
  status: "dark-outline"
};
var ButtonRounded = Template.bind({});
exports.ButtonRounded = ButtonRounded;
ButtonRounded.args = {
  label: "Button Rounded",
  isRounded: true
};
var ButtonSmall = Template.bind({});
exports.ButtonSmall = ButtonSmall;
ButtonSmall.args = {
  label: "Button Small",
  size: "small"
};
var ButtonNormal = Template.bind({});
exports.ButtonNormal = ButtonNormal;
ButtonNormal.args = {
  label: "Button Normal"
};
var ButtonLarge = Template.bind({});
exports.ButtonLarge = ButtonLarge;
ButtonLarge.args = {
  label: "Button Large",
  size: "large"
};
var ButtonDisabled = Template.bind({});
exports.ButtonDisabled = ButtonDisabled;
ButtonDisabled.args = {
  label: "Button Disabled",
  disabled: true
};
var ButtonIcon = Template.bind({});
exports.ButtonIcon = ButtonIcon;
ButtonIcon.args = {
  label: "Button Icon",
  icon: /*#__PURE__*/_react.default.createElement(_Icon.default, {
    name: "bi bi-alarm-fill"
  })
};
var ButtonGroup = ButtonGroupWrapper.bind({});
exports.ButtonGroup = ButtonGroup;
ButtonGroup.args = {
  children: /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_Button.default, {
    label: "left"
  }), /*#__PURE__*/_react.default.createElement(_Button.default, {
    label: "middle"
  }), /*#__PURE__*/_react.default.createElement(_Button.default, {
    label: "right"
  }))
};
