// react-mazer-ui/dist/Card.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CardWithHeaderFooter = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _Card = _interopRequireDefault(require("./Card"));

var _CardBody = _interopRequireDefault(require("./CardBody"));

var _CardHeader = _interopRequireDefault(require("./CardHeader"));

var _CardFooter = _interopRequireDefault(require("./CardFooter"));

var _Button = _interopRequireDefault(require("./Button"));

var _Image = _interopRequireDefault(require("./Image"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'Example/Card',
  component: _Card.default,
  argTypes: {}
};
exports.default = _default;

var Template = function Template(args) {
  return /*#__PURE__*/_react.default.createElement(_Card.default, args);
};

var CardWithHeaderFooter = Template.bind({});
exports.CardWithHeaderFooter = CardWithHeaderFooter;
CardWithHeaderFooter.args = {
  children: /*#__PURE__*/_react.default.createElement("div", null, /*#__PURE__*/_react.default.createElement(_CardBody.default, null, /*#__PURE__*/_react.default.createElement(_CardHeader.default, {
    title: "Card With Header And Footer"
  }), /*#__PURE__*/_react.default.createElement("p", null, "Gummies bonbon apple pie fruitcake icing biscuit apple pie jelly-o sweet roll. Toffee sugar plum sugar plum jelly-o jujubes bonbon dessert carrot cake."), /*#__PURE__*/_react.default.createElement(_Image.default, {
    source: "https://zuramai.github.io/mazer/demo/assets/images/samples/banana.jpg"
  })), /*#__PURE__*/_react.default.createElement(_CardFooter.default, {
    className: "d-flex justify-content-between"
  }, /*#__PURE__*/_react.default.createElement("span", null, "Card Footer"), /*#__PURE__*/_react.default.createElement(_Button.default, {
    label: "Read More",
    status: "primary"
  })))
};
