// react-mazer-ui/dist/Dropdown.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.DropdownDark = exports.DropdownWarning = exports.DropdownDanger = exports.DropdownInfo = exports.DropdownSuccess = exports.DropdownSecondary = exports.DropdownPrimary = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _Dropdown = _interopRequireDefault(require("./Dropdown"));

var _DropdownToggle = _interopRequireDefault(require("./DropdownToggle"));

var _DropdownMenu = _interopRequireDefault(require("./DropdownMenu"));

var _DropdownItem = _interopRequireDefault(require("./DropdownItem"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'Example/Dropdown',
  component: _Dropdown.default,
  argTypes: {},
  parameters: {
    height: {
      values: 300
    }
  }
};
exports.default = _default;

var Template = function Template(args) {
  return /*#__PURE__*/_react.default.createElement(_Dropdown.default, args);
};

var DropdownPrimary = Template.bind({});
exports.DropdownPrimary = DropdownPrimary;
DropdownPrimary.args = {
  children: /*#__PURE__*/_react.default.createElement("div", null, /*#__PURE__*/_react.default.createElement(_DropdownToggle.default, {
    id: "dropdown-1",
    status: "primary"
  }, "Primary"), /*#__PURE__*/_react.default.createElement(_DropdownMenu.default, null, /*#__PURE__*/_react.default.createElement(_DropdownItem.default, null, "Menu A"), /*#__PURE__*/_react.default.createElement(_DropdownItem.default, null, "Menu B"), /*#__PURE__*/_react.default.createElement(_DropdownItem.default, null, "Menu C"), /*#__PURE__*/_react.default.createElement(_DropdownItem.default, null, "Menu D")))
};
var DropdownSecondary = Template.bind({});
exports.DropdownSecondary = DropdownSecondary;
DropdownSecondary.args = {
  children: /*#__PURE__*/_react.default.createElement("div", null, /*#__PURE__*/_react.default.createElement(_DropdownToggle.default, {
    id: "dropdown-1",
    status: "secondary"
  }, "Secondary"), /*#__PURE__*/_react.default.createElement(_DropdownMenu.default, null, /*#__PURE__*/_react.default.createElement(_DropdownItem.default, null, "Menu A"), /*#__PURE__*/_react.default.createElement(_DropdownItem.default, null, "Menu B"), /*#__PURE__*/_react.default.createElement(_DropdownItem.default, null, "Menu C"), /*#__PURE__*/_react.default.createElement(_DropdownItem.default, null, "Menu D")))
};
var DropdownSuccess = Template.bind({});
exports.DropdownSuccess = DropdownSuccess;
DropdownSuccess.args = {
  children: /*#__PURE__*/_react.default.createElement("div", null, /*#__PURE__*/_react.default.createElement(_DropdownToggle.default, {
    id: "dropdown-1",
    status: "success"
  }, "Success"), /*#__PURE__*/_react.default.createElement(_DropdownMenu.default, null, /*#__PURE__*/_react.default.createElement(_DropdownItem.default, null, "Menu A"), /*#__PURE__*/_react.default.createElement(_DropdownItem.default, null, "Menu B"), /*#__PURE__*/_react.default.createElement(_DropdownItem.default, null, "Menu C"), /*#__PURE__*/_react.default.createElement(_DropdownItem.default, null, "Menu D")))
};
var DropdownInfo = Template.bind({});
exports.DropdownInfo = DropdownInfo;
DropdownInfo.args = {
  children: /*#__PURE__*/_react.default.createElement("div", null, /*#__PURE__*/_react.default.createElement(_DropdownToggle.default, {
    id: "dropdown-1",
    status: "info"
  }, "Info"), /*#__PURE__*/_react.default.createElement(_DropdownMenu.default, null, /*#__PURE__*/_react.default.createElement(_DropdownItem.default, null, "Menu A"), /*#__PURE__*/_react.default.createElement(_DropdownItem.default, null, "Menu B"), /*#__PURE__*/_react.default.createElement(_DropdownItem.default, null, "Menu C"), /*#__PURE__*/_react.default.createElement(_DropdownItem.default, null, "Menu D")))
};
var DropdownDanger = Template.bind({});
exports.DropdownDanger = DropdownDanger;
DropdownDanger.args = {
  children: /*#__PURE__*/_react.default.createElement("div", null, /*#__PURE__*/_react.default.createElement(_DropdownToggle.default, {
    id: "dropdown-1",
    status: "danger"
  }, "Danger"), /*#__PURE__*/_react.default.createElement(_DropdownMenu.default, null, /*#__PURE__*/_react.default.createElement(_DropdownItem.default, null, "Menu A"), /*#__PURE__*/_react.default.createElement(_DropdownItem.default, null, "Menu B"), /*#__PURE__*/_react.default.createElement(_DropdownItem.default, null, "Menu C"), /*#__PURE__*/_react.default.createElement(_DropdownItem.default, null, "Menu D")))
};
var DropdownWarning = Template.bind({});
exports.DropdownWarning = DropdownWarning;
DropdownWarning.args = {
  children: /*#__PURE__*/_react.default.createElement("div", null, /*#__PURE__*/_react.default.createElement(_DropdownToggle.default, {
    id: "dropdown-1",
    status: "warning"
  }, "Warning"), /*#__PURE__*/_react.default.createElement(_DropdownMenu.default, null, /*#__PURE__*/_react.default.createElement(_DropdownItem.default, null, "Menu A"), /*#__PURE__*/_react.default.createElement(_DropdownItem.default, null, "Menu B"), /*#__PURE__*/_react.default.createElement(_DropdownItem.default, null, "Menu C"), /*#__PURE__*/_react.default.createElement(_DropdownItem.default, null, "Menu D")))
};
var DropdownDark = Template.bind({});
exports.DropdownDark = DropdownDark;
DropdownDark.args = {
  children: /*#__PURE__*/_react.default.createElement("div", null, /*#__PURE__*/_react.default.createElement(_DropdownToggle.default, {
    id: "dropdown-1",
    status: "dark"
  }, "Dark"), /*#__PURE__*/_react.default.createElement(_DropdownMenu.default, null, /*#__PURE__*/_react.default.createElement(_DropdownItem.default, null, "Menu A"), /*#__PURE__*/_react.default.createElement(_DropdownItem.default, null, "Menu B"), /*#__PURE__*/_react.default.createElement(_DropdownItem.default, null, "Menu C"), /*#__PURE__*/_react.default.createElement(_DropdownItem.default, null, "Menu D")))
};
