// @storybook/addon-storyshots/dist/ts3.9/frameworks/angular/renderTree.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __spreadArrays = (this && this.__spreadArrays) || function () {
    for (var s = 0, i = 0, il = arguments.length; i < il; i++) s += arguments[i].length;
    for (var r = Array(s), k = 0, i = 0; i < il; i++)
        for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
            r[k] = a[j];
    return r;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var AngularSnapshotSerializer_1 = __importDefault(require("jest-preset-angular/build/AngularSnapshotSerializer"));
var HTMLCommentSerializer_1 = __importDefault(require("jest-preset-angular/build/HTMLCommentSerializer"));
var testing_1 = require("@angular/core/testing");
var testing_2 = require("@angular/platform-browser-dynamic/testing");
var jest_specific_snapshot_1 = require("jest-specific-snapshot");
var renderer_1 = require("@storybook/angular/renderer");
var rxjs_1 = require("rxjs");
jest_specific_snapshot_1.addSerializer(HTMLCommentSerializer_1.default);
jest_specific_snapshot_1.addSerializer(AngularSnapshotSerializer_1.default);
function getRenderedTree(story) {
    var currentStory = story.render();
    var moduleMeta = renderer_1.getStorybookModuleMetadata({
        storyFnAngular: currentStory,
        parameters: story.parameters,
        // TODO : To change with the story Id in v7. Currently keep with static id to avoid changes in snapshots
        targetSelector: 'storybook-wrapper',
    }, new rxjs_1.BehaviorSubject(currentStory.props));
    testing_1.TestBed.configureTestingModule({
        imports: __spreadArrays(moduleMeta.imports),
        declarations: __spreadArrays(moduleMeta.declarations),
        providers: __spreadArrays(moduleMeta.providers),
        schemas: __spreadArrays(moduleMeta.schemas),
    });
    testing_1.TestBed.overrideModule(testing_2.BrowserDynamicTestingModule, {
        set: {
            entryComponents: __spreadArrays(moduleMeta.entryComponents),
        },
    });
    return testing_1.TestBed.compileComponents().then(function () {
        var tree = testing_1.TestBed.createComponent(moduleMeta.bootstrap[0]);
        tree.detectChanges();
        // Empty componentInstance remove attributes of the internal main component (<storybook-wrapper>) in snapshot
        return __assign(__assign({}, tree), { componentInstance: {} });
    });
}
exports.default = getRenderedTree;
